cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/config/BrokerSetFileResolver.java

```diff
@@ -59,12 +59,25 @@ public Map<String, Set<Integer>> brokerIdsByBrokerSetId(ClusterModel clusterMode
     try {
       brokerIdsByBrokerSetId = loadBrokerSetData();
     } catch (IOException e) {
-      throw new IllegalArgumentException(e);
+      throw new BrokerSetResolutionException(e.getMessage());
     }
 
     return _brokerSetAssignmentPolicy.assignBrokerSetsForUnresolvedBrokers(clusterModel, brokerIdsByBrokerSetId);
   }
 
+  @Override
+  public Map<String, Set<Integer>> brokerIdsByBrokerSetId(Map<Integer, String> rackIdByBrokerId)
+      throws BrokerSetResolutionException {
+    Map<String, Set<Integer>> brokerIdsByBrokerSetId;
+    try {
+      brokerIdsByBrokerSetId = loadBrokerSetData();
+    } catch (IOException e) {
+      throw new BrokerSetResolutionException(e.getMessage());
```

Comment:
Why not handle it gracefully if the file is missing / malformed? I think this method should be robust, and rely on the `assignBrokerSetsForUnresolvedBrokers` to handle the error scenario.

Reply:
I think we intentionally want to remind the user that their config file is missing. E.g. they may put the file in wrong place, and in that case we don't want the user to mis-think that BrokerSetResolver is using their json file.